mobilecoinofficial/full-service
# Ensure comments actually update in a transaction log

### Motivation

When a field is provided in an API, we expect that field to be leveraged. Specifically, in this case, the `submit_transaction` API wasn't using the provided comment to update the comment field in an existing transaction log. 

### In this PR
We update the comment both when signing the transaction, and when submitting it. 

### Test Plan
To run the rust test suite (including the test added in this PR), copy the consensus_enclave.css file to your current working directory, and then run `tools/test.sh`

Manually, you can exercise the scenario by doing the following

1. Create a transaction with the `build_transaction` API
2. Grab the `transaction_log_id` from the response
3. Check the transaction log with the `get_transaction_log` API using the `transaction_log_id` from Step 2
4. Observe that the comment field in `get_transaction_log`'s response is an empty string
5. Submit the transaction using the `submit_transaction` API, and specify the optional parameter `comment` to be a string of your choice
6. Check the transaction log with the `get_transaction_log` API using the `transaction_log_id` from Step 2 and observe your new comment!
